File: validateCreditCardNumber/validateCreditCardNumber.js
function validate(n){
  var numString = n.toString();
  var numlength = numString.length;
  var myArray = numString
  for(var i=0; i<myArray.length; i++) { myArray[i] = parseInt(myArray[i], 10); }
  console.log("MyArray: ", myArray[2]);
//   var b = n.split('').map(Number);
  if (numlength % 2 == 0)
  {
    var index = 0;
    while (numString[index])
    {
      var replaceNum = parseInt(numString[index]);
      replaceNum = replaceNum * 2;
      replaceNum = replaceNum.toString();
      numString[index] = replaceNum;
      index+=2;
    }
  }
  else
  {
  }
  console.log(numlength);
  console.log(numString[1]);
}

File: validateCreditCardNumber/validateCreditCardNumber.java
public class Validate{
  public static boolean validate(String n){
    // TODO: solution 
    String[] integerStrings = n.split(""); 
    Integer[] intarray=new Integer[integerStrings.length];
    int index=0;
    int total = 0;
          System.out.println(integerStrings.length);
    if (n.length() % 2 == 0)
    {
      for(String str:integerStrings){
          intarray[index]=Integer.parseInt(str.trim());//get rid of additional whitespace
          if (index % 2 == 0)
          {
            intarray[index] = intarray[index] * 2;
            if (intarray[index] > 9)
              intarray[index] = intarray[index] - 9;
          }
          else
          {
            intarray[index] = intarray[index];
          }
          total = intarray[index] + total;
          System.out.println("Part 1");
          System.out.println(intarray[index]);
          index++;
      }
    }
    else
    {
      for(String str:integerStrings){
          intarray[index]=Integer.parseInt(str.trim());//get rid of additional whitespace
          if (index % 2 != 0)
          {
            intarray[index] = intarray[index] * 2;
            if (intarray[index] > 9)
              intarray[index] = intarray[index] - 9;
          }
          total = intarray[index] + total;
          System.out.println("Part 2");
          System.out.println(intarray[index]);
          index++;
      }
    }
    if (total % 10 == 0)
      return true;
    else
      return false;
  }
}